packages/shylu/shylu_node/fastilu/src/shylu_fastilu.hpp

```diff
@@ -528,145 +935,103 @@ class FastILUPrec
         {
             using WithoutInit = Kokkos::ViewAllocateWithoutInitializing;
+            const Ordinal nRowsUnblocked = pRowMap_.extent(0) - 1;
             Ordinal nnzA = 0;
-            for (Ordinal k = 0; k < pRowMap_(nRows); k++)  {
+            for (Ordinal k = 0; k < pRowMap_(nRowsUnblocked); k++)  {
                 if(pLvlIdx_(k) <= level) {
                    nnzA++;
                 }
             }
             //Initialize the A matrix that is to be used in the computation
-            aRowMap = OrdinalArray(WithoutInit("aRowMap"), nRows + 1);
-            aColIdx = OrdinalArray(WithoutInit("aColIdx"), nnzA);
-            aRowIdx = OrdinalArray(WithoutInit("aRowIds"), nnzA);
-            aRowMap_ = Kokkos::create_mirror_view(Kokkos::WithoutInitializing, aRowMap);
-            aColIdx_ = Kokkos::create_mirror_view(Kokkos::WithoutInitializing, aColIdx);
-            aRowIdx_ = Kokkos::create_mirror_view(Kokkos::WithoutInitializing, aRowIdx);
-
-            aVal = ScalarArray(WithoutInit("aVal"), nnzA);
-            aVal_ = Kokkos::create_mirror_view(Kokkos::WithoutInitializing, aVal);
+            aRowMap_ = OrdinalArrayMirror(WithoutInit("aRowMap"), nRowsUnblocked + 1);
+            aColIdx_ = OrdinalArrayMirror(WithoutInit("aColIdx"), nnzA);
+            aRowIdx_ = OrdinalArrayMirror(WithoutInit("aRowIds"), nnzA);
+            aVal_    = ScalarArrayMirror(WithoutInit("aVal"), nnzA);
 
             Ordinal aRowPtr = 0;
             aRowMap_[0] = aRowPtr;
-            for (Ordinal i = 0; i < nRows; i++) 
+            for (Ordinal i = 0; i < nRowsUnblocked; i++)
             {
                 for(Ordinal k = pRowMap_(i); k < pRowMap_(i+1); k++)
                 {
                     if (pLvlIdx_(k) <= level) {
-                        aVal_[aRowPtr] = pVal_[k];
                         aColIdx_[aRowPtr] = pColIdx_[k];
                         aRowIdx_[aRowPtr] = i;
                         aRowPtr++;
                     }
                 }
                 aRowMap_[i+1] = aRowPtr;
             }
-            // sort based on ColIdx, RowIdx stays the same (do we need this?)
-            //using host_space = Kokkos::HostSpace::execution_space;
-            //KokkosSparse::sort_crs_matrix<host_space, OrdinalArrayMirror, OrdinalArrayMirror, ScalarArrayMirror>
-            //  (aRowMap_, aColIdx_, aVal_);
-            //host_space().fence();
+        }
+
+        void symbolicILU_common()
+        {
+            using WithoutInit = Kokkos::ViewAllocateWithoutInitializing;
+
+            // Ensure all filled entries have the sentinel value
+#ifdef FASTILU_ONE_TO_ONE_UNBLOCKED
+            aVal_ = ScalarArrayMirror("aVal", aColIdx_.extent(0));
+#else
+            aVal_ = ScalarArrayMirror("aVal", aColIdx_.extent(0) * blockCrsSize * blockCrsSize);
+#endif
+            Kokkos::deep_copy(aVal_, std::numeric_limits<Scalar>::min());
```

Comment:
Min is used to denote values that FastILUFunctor should not skip (the fills). FastILUFunctor will skip true zeroes.